extractors/mysql/extractor.go

```diff
@@ -0,0 +1,145 @@
+package mysql
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+
+	"github.com/mitchellh/mapstructure"
+)
+
+type Extractor struct{}
+
+type Config struct {
+	UserID   string `mapstructure:"user_id"`
+	Password string `mapstructure:"password"`
+	Host     string `mapstructure:"host"`
+}
+
+var defaultDBList = []string{
+	"information_schema",
+	"mysql",
+	"performance_schema",
+	"sys",
+}
+
+func (e *Extractor) Extract(configMap map[string]interface{}) (result []map[string]interface{}, err error) {
+	config, err := e.getConfig(configMap)
+	if err != nil {
+		return
+	}
+	err = e.validateConfig(config)
+	if err != nil {
+		return
+	}
+	db, err := sql.Open("mysql", fmt.Sprintf("%s:%s@tcp(%s)/", config.UserID, config.Password, config.Host))
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	result, err = getDatabases(db)
+	return
+}
+
+func getDatabases(db *sql.DB) (result []map[string]interface{}, err error) {
+	res, err := db.Query("SHOW DATABASES;")
+	if err != nil {
+		return
+	}
+	for res.Next() {
+		var database string
+		res.Scan(&database)
+		if checkNotDefaultDatabase(database) {
+			result, _ = tableInfo(database, result, db)
+		}
+	}
+	return
+}
+
+func tableInfo(dbName string, result []map[string]interface{}, db *sql.DB) (_ []map[string]interface{}, err error) {
```

Comment:
we could change the method name to  represent what it does clearly.
right now it receives a list (result) and populate it with another table information.

so probably we can change it to `populateWithTables`.
wdyt?

Reply:
I think ```getTablesInfo``` will be a better name, actually it takes in a dbName, and appends the information of all the tables in that db, but since we are dealing with db's and it performs a ```SELECT``` operation and not ```INSERT``` I think populateWithTables might be confusing??